apps/public-docsite/src/components/IconGrid/IconGrid.tsx

```diff
@@ -1,40 +1,98 @@
 import * as React from 'react';
-import { Icon, SearchBox } from '@fluentui/react';
+import { FontIcon, ISearchBoxProps, SearchBox, Spinner } from '@fluentui/react';
 import * as stylesImport from './IconGrid.module.scss';
 const styles: any = stylesImport;
 
-export interface IIconGridProps {
+export interface IFontIconGridProps {
   /**
-   * An array of icons.
+   * An array of font-based icon names.
    */
   icons: { name: string }[];
 
   /**
-   * If we should render using `Icon` from Fabric
+   * Type of icons:
+   * - `core-font` is Fabric Core font icons, rendered in an `<i>` tag with classes
+   * - `react-font` is Fluent UI React font icons, rendered using `<FontIcon>`
+   * - `react-svg` is SVG icon components, rendered directly
    */
-  useFabricIcons?: boolean;
+  iconType: 'core-font' | 'react-font';
 }
 
-export interface IIconGridState {
+export interface ISvgIconGridProps {
+  /**
+   * A promise loading a package of SVG icon components.
+   */
+  icons: Promise<{ [exportName: string]: any }>;
+
+  iconType: 'react-svg';
+}
+
+export type IIconGridProps = IFontIconGridProps | ISvgIconGridProps;
+
+interface IFontIconDefinition {
+  name: string;
+  ref: React.RefObject<HTMLElement>;
+  iconType: 'core-font' | 'react-font';
+}
+
+interface ISvgIconDefinition {
+  name: string;
+  component: React.ComponentType;
+  iconType: 'react-svg';
+}
+
+type IIconDefinition = IFontIconDefinition | ISvgIconDefinition;
+
+interface IIconGridState {
   /**
    * The text we are filtering the icons by.
    */
   searchQuery: string;
+
+  /**
+   * The actual icons to show. This is generated immediately for a static list of font-based icons,
+   * or generated on load for a list of SVG icon components.
+   */
+  resolvedIcons?: IIconDefinition[];
 }
 
 export class IconGrid extends React.Component<IIconGridProps, IIconGridState> {
-  private _iconRefs: { [iconName: string]: React.RefObject<HTMLElement> };
-
   constructor(props: IIconGridProps) {
     super(props);
 
-    this.state = {
+    const state: IIconGridState = {
       searchQuery: '',
     };
 
-    this._iconRefs = {};
-    for (const icon of props.icons) {
-      this._iconRefs[icon.name] = React.createRef();
+    if (props.iconType !== 'react-svg') {
```

Comment:
while adding/refactoring, shouldn't we refactor this to function component with hooks to lower our technical depth ?

Reply:
This website is basically in maintenance mode, so I wouldn't consider function component conversions to be a priority (if we're going to spend any time on improvements here, there are other issues that would be much higher impact/priority). I guess with the level of rewriting I did here, I could have also converted to a function component, but at this point it's probably not worth the time to go back and do that.

Reply:
I see, well for me it would be easier to directly rewrite to function component heh. regarding maintenance mode - do we have End Of Life date ?